### DISCUSSION TITLE
Longitudinal functional MRI - multi-step registration

### DISCUSSION THREAD

**Discussion by mclooned** [CoBrALab/RABIES#350] [Q&A] [answered]
Hi all,

We are working with a longitudinal fMRI dataset in rats. Our registration is a multi-step process which involves registering a subject’s anatomical scan to that subject’s average template (a template constructed from scans taken at different time points). This is then registered to a population average template. Is it possible to integrate these steps into the RABIES pipeline?

Thanks,
-Daniel

**Comment by Gab-D-G** [comment, marked as answer]
Hi Daniel,

Thank you for the input, this would be indeed ideal for longitudinal experiments, and it would be great to have this as an option within RABIES. I have opened a new issue on this topic https://github.com/CoBrALab/RABIES/issues/352. We would welcome contributions in implementing this new option. Feel free to consult our guidelines in this regard https://rabies.readthedocs.io/en/latest/contributing.html, and I can provide assistance.

Best,
Gabriel DG

**Reply by mclooned** [reply]
Thanks Gabriel, 

Apologies for the delayed reply: we had a few deadlines this week. That sounds good to me, we will start working on trying to implement this from next week on. I'll likely be in touch in the next few weeks.

Best,
Daniel

**Reply by Gab-D-G** [reply]
Hi Daniel, upon revision of the issue, I realize the changes required to the current workflow may be substantial, and require a good understanding of nipype workflows, and the current workflow architecture within RABIES. I still think it would be a great addition, and something that should be possible in the future, but I cannot promise I will come to implement it myself at the moment. It is not a beginner's task, but do let me know if you want to give it a serious try and have questions for me.

**Reply by amclarekelly** [reply]
Hi Gabriel!
Thanks for looking into this and proposing it as task the community might work on.
We don't quite have the coding capacity here to do this in a flexible/sharable way right now and are looking for a fix that might get us to where we need to go in the short term. 
One thing we will try is to create our group-specific template outside the rabies pipeline, and use that as the template for the main processing run. 
But in our experience, registration fails for the youngest rats, who are much smaller than the oldest. If this continues to be a problem, do you think it would be feasible to compute the two-step registration outside the rabies pipeline, but then replace the rabies-computed transformation (outputs of rabies.preprocess_pkg.commonspace_reg.init_commonspace_reg_wf) with our two-step transformation? Would it be possible to do this manual intervention, then have the preprocessing run as "usual"?
thanks so much for your help!
Clare

**Reply by Gab-D-G** [reply]
Hi Clare, it would definitely be possible in theory to compute your own two level registration and run RABIES on top of it. 

Option 1: This could be done by obtaining your templates for each of your sub-groups (e.g. the template for your young rats) outside of RABIES, then running RABIES on that subgroup while providing the template for that group to replace the commonspace files (i.e. --anat_template, etc...), and by using the fast_commonspace option from --commonspace_reg, which will effectively skip the generation of the intermediate template within RABIES and directly register each scan onto  your pre-generated template.

Option 2: if you have already registered the anatomical scans for each of your subject with your two-level registration, you could then in principle run RABIES on each EPI scan individual with the --bold_only option and provide the preprocessed anatomical scan for that subject, and using fast_commonspace as above. The EPI will then be registered directly onto the anatomical scan. Importantly, the anatomical scan should be corrected for intensity inhomogeneities, otherwise registration won't work. 

An important caveat however is that in both cases the preprocessed timeseries are resampled onto the template you provide, so if you want all timeseries to be resampled onto a different commonspace, you will need to resample them yourself by combining the transforms you computed with those computed by RABIES. Let me know if you have questions in this regard.

Also, you will need to provide anatomical masks and labels along with the --anat_template, but you should be able to derive those from your commonspace template once you've run your own registration.

I hope this helps! Let me know if further clarifications are needed.

Gabriel
